### PR TITLE
Implement internal/net/grpc metrics & proto & types test

### PR DESCRIPTION
**Pull request by kevindiu**
Signed-off-by: kevindiu <kevindiujp@gmail.com>

This PR implements internal/net/grpc metrics & proto & types test.
### Environment:

- Go Version: 1.17.3
- Docker Version: 20.10.8
- Kubernetes Version: 1.22.0
- NGT Version: 1.13.8

### Types of changes:

- [ ] Bug fix [type/bug]
- [ ] New feature [type/feature]
- [x] Add tests [type/test]
- [ ] Security related changes [type/security]
- [ ] Add documents [type/documentation]
- [ ] Refactoring [type/refactoring]
- [ ] Update dependencies [type/dependency]
- [ ] Update benchmarks and performances [type/bench]
- [ ] Update CI [type/ci]

### Changes to Core Features:

- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- [x] Have you written new tests for your core changes, as applicable?
- [x] Have you successfully ran tests with your changes locally?

### Checklist:

- [x] I have read the [CONTRIBUTING](https://github.com/vdaas/vald/blob/master/CONTRIBUTING.md) document.
- [x] I have checked open [Pull Requests](https://github.com/vdaas/vald/pulls) for the similar feature or fixes?
- [x] I have added tests and benchmarks to cover my changes.
- [x] I have ensured all new and existing tests passed.
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: internal/net/grpc/proto/proto_test.go [hunk 1]
```diff
@@ -21,10 +21,181 @@ import (
...
+			}
+			if test.checkFunc == nil {
+				test.checkFunc = defaultCheckFunc
+			}
+
+			err := Unmarshal(test.args.data, test.args.v)
+			if err := test.checkFunc(test, err); err != nil {
+				tt.Errorf("error = %v", err)
+			}
+
```
[the remark this answers is not in the document]
Comment (kevindiu) [reply internal/net/grpc/proto/proto_test.go:194]:
```suggestion
```

File: internal/net/grpc/proto/proto_test.go [hunk 2]
```diff
@@ -21,10 +21,181 @@ import (
...
+			}
+			if test.checkFunc == nil {
+				test.checkFunc = defaultCheckFunc
+			}
+
+			got, err := Marshal(test.args.m)
+			if err := test.checkFunc(test.want, got, err); err != nil {
+				tt.Errorf("error = %v", err)
+			}
+
```
[the remark this answers is not in the document]
Comment (kevindiu) [reply internal/net/grpc/proto/proto_test.go:100]:
```suggestion
```

File: internal/net/grpc/metric/client_test.go [hunk 3]
```diff
@@ -48,31 +48,15 @@ func TestNewClientHandler(t *testing.T) {
...
-		*/
+		{
+			name: "return client handler",
+			args: args{
+				opts: nil,
+			},
+			want: want{
+				want: new(ClientHandler),
+			},
+		},
```
Comment (vankichi) [inline internal/net/grpc/metric/client_test.go:59]: please add the case of `opts is not nil`

File: internal/net/grpc/metric/server_test.go [hunk 4]
```diff
@@ -48,31 +48,12 @@ func TestNewServerHandler(t *testing.T) {
...
-		           checkFunc: defaultCheckFunc,
-		       }
-		   }(),
-		*/
+		{
+			name: "return server handler",
+			want: want{
+				want: new(ServerHandler),
+			},
+		},
```
Comment (vankichi) [inline internal/net/grpc/metric/server_test.go:56]: same

File: internal/net/grpc/proto/proto_test.go [hunk 5]
```diff
@@ -21,10 +21,179 @@ import (
...
+				data: []byte{1, 2, 3, 4},
+				v:    &payload.Object_Vector{},
+			},
+			checkFunc: func(t test, e error) error {
+				if e == nil {
+					return errors.New("error should be returned")
+				}
+				return nil
+			},
+		},
```
Comment (vankichi) [inline internal/net/grpc/proto/proto_test.go:171]: please use same table format

**Review by kmrmt — DISMISSED** [review]
I have no other comment than what is already pointed.

File: internal/net/grpc/metric/client_test.go (hunk 3 above)
Comment (kevindiu) [reply internal/net/grpc/metric/client_test.go:60]: fixed

File: internal/net/grpc/proto/proto_test.go (hunk 5 above)
Comment (kevindiu) [reply internal/net/grpc/proto/proto_test.go]: fixed

File: internal/net/grpc/metric/server_test.go (hunk 4 above)
Comment (kevindiu) [reply internal/net/grpc/metric/server_test.go:56]: fixed

**Review by vankichi — DISMISSED** [review]

**Review by kmrmt — APPROVED** [review]
LGTM

**Pull request merged**